pkg/workflow/workflow.go

```diff
@@ -355,8 +355,19 @@ func (e *engine) getBackoffWaitTime() int {
 	// the default value of min times reaches the max workflow backoff wait time
 	minTimes := 15
 	found := false
+
+	var stepIds []string
 	for _, step := range e.status.Steps {
-		if v, ok := e.wfCtx.GetValueInMemory(wfTypes.ContextPrefixBackoffTimes, step.ID); ok {
+		if step.SubStepsStatus == nil {
+			stepIds = append(stepIds, step.ID)
+		} else {
+			for _, subStep := range step.SubStepsStatus {
+				stepIds = append(stepIds, subStep.ID)
+			}
+		}
+	}
+	for _, stepID := range stepIds {
+		if v, ok := e.wfCtx.GetValueInMemory(wfTypes.ContextPrefixBackoffTimes, stepID); ok {
```

Comment:
We can make this a function like `getMinTimes`, in that way, we can range once for `e.status.Steps`.

Reply:
Ok